### PR TITLE
WIP: Search ISS and display results

### PR DESCRIPTION
**Pull request by danni**
Todo:
- Needs tests
- Needs styling
- Needs per category information

### CODE REVIEW

**Comment by DanielHeath** [comment]
https://gist.github.com/DanielHeath/cb2f182006486f8f2f0e

File: src/pages/CategoryPage.js [hunk 1]
```diff
@@ -9,19 +36,54 @@ class CategoryPage extends React.Component {
...
+        });
+
+        return JSON.parse(response.text);
     }
 
-    componentWillReceiveProps(nextProps: Object): void {
-        this.setState({ categoryKey: nextProps.params.categoryName });
+    componentDidMount(): void {
+        this.search()
+            .then(data => {
```
Comment (DanielHeath) [inline src/pages/CategoryPage.js:47]: Error handling?

File: src/components/CategoryListItem.scss [hunk 2]
```diff
@@ -1,5 +1,3 @@
-@import '../styles/colors.scss';
```
Comment (DanielHeath) [inline src/components/CategoryListItem.scss:1]: How is $icon-fg-color injected if we aren't requiring the colors?

File: src/components/HeaderBar.scss [hunk 3]
```diff
@@ -1,5 +1,3 @@
-@import '../styles/colors.scss';
```
Comment (DanielHeath) [inline src/components/HeaderBar.scss]: How is $icon-fg-color injected if we aren't requiring the colors?

File: src/components/ResultTile.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            {}
+        );
+    }
+
+    render(): React.Element {
+        var {
+            object,
+        } = this.props;
+
+        console.log(object);
```
Comment (DanielHeath) [inline src/components/ResultTile.js:39]: Logging?

File: src/components/ResultTile.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+.ResultTile {
+    .colored-icon path {
+        fill: $icon-fg-color;
```
Comment (DanielHeath) [inline src/components/ResultTile.scss:3]: Where does this color come from?

File: src/pages/CategoryPage.js [hunk 6]
```diff
@@ -1,27 +1,70 @@
...
+            limit: 3,
+        })
+            .then(data => {
+                this.setState({
+                    meta: data.meta,
+                    objects: data.objects,
+                });
+            });
+
+        // FIXME: display error
```
Comment (DanielHeath) [inline src/pages/CategoryPage.js:54]: Put this fixme inside a `.catch(error) => {`

File: src/pages/ServicePage.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+     * search:
+     * Do a search in ISS
+     *
+     * Returns: a promise to an object of data from ISS.
+     */
+
+    componentDidMount(): void {
+        iss(`service/${this.props.params.id}/`)
+            .then(data => {
+                console.log(data);
```
Comment (DanielHeath) [inline src/pages/ServicePage.js:29]: Logging

File: src/pages/ServicePage.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * search:
+     * Do a search in ISS
+     *
+     * Returns: a promise to an object of data from ISS.
+     */
+
+    componentDidMount(): void {
+        iss(`service/${this.props.params.id}/`)
```
Comment (DanielHeath) [inline src/pages/ServicePage.js:27]: I'd prefer to create a 'loading' component which shows errors consistently, etc - then make all the other components synchronous.

File: src/components/CategoryListItem.scss (hunk 2 above)
Comment (danni) [inline src/components/CategoryListItem.scss:1]: It's imported into the main components.scss. Imports happen before the compile.

File: src/components/ResultTile.scss (hunk 5 above)
Comment (danni) [inline src/components/ResultTile.scss:3]: components.scss

File: src/pages/ServicePage.js (hunk 8 above)
Comment (danni) [inline src/pages/ServicePage.js:27]: Yeah, I had the same idea.

**Pull request merged**